app/controllers/api/functions.php

```diff
@@ -2356,3 +2356,39 @@
 
         $response->noContent();
     });
+
+App::get('/v1/functions/templates')
+    ->desc('List function templates')
+    ->label('scope', 'public')
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'listFunctionTemplates')
+    ->label('sdk.description', '/docs/references/functions/list-function-templates.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_TEMPLATE_FUNCTION_LIST)
+    ->param('runtimes', [], new ArrayList(new WhiteList(array_keys(Config::getParam('runtimes')), true), APP_LIMIT_ARRAY_PARAMS_SIZE), 'List of runtimes allowed for filtering function templates. Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' runtimes are allowed.', true)
```

Comment:
I think we also need filter for ID or IDs. So details page don't need to load everything

Reply:
that should be a different endpoint `GET /v1/functions/template/:templateId` so not required for now.

Reply:
Lets have getTempalte endpoint then. Or should we do this listTempaltes on details page of a template? it feels like big big overkill to load all the templates here:
![CleanShot 2024-07-30 at 12 47 08@2x](https://github.com/user-attachments/assets/a6ceff42-6cf0-4a1e-8920-e57f3995c682)